### PR TITLE
Update README to update the branch details

### PR DESCRIPTION
**Pull request by leegeth**
*Description of changes:*
Update README to update the branch details. The changes include
1. Update the branches descriptions in README to reflect the status after release.
2. Mention about the tag based releases.
3. Description about development branch needs to be removed after merging development branch to master.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
 
-## Development branch
+## Branches
+
+
+### master branch
+The [master](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master) branch hosts the continuous development of the AWS IoT Embedded C SDK libraries. Please be aware that the libraries in master branch may have bugs and performance issues. Consider using the released versions of the AWS IoT Embedded C SDK for production ready software.
+
+### v4_beta branch
+The [v4_beta](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v4_beta) branch contains a beta version of the AWS IoT Embedded C SDK libraries. The libraries in this branch will not be released. However, critical bugs will be fixed and tested. No new features will be added to this branch.
```
Comment (abhidixi11) [inline README.md:12]: We should rename the v4_beta branch.

**Review by yngki — DISMISSED** [review]

**Comment by leegeth** [comment]
Reopening the PR on master branch.

**Review by abhidixi11 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
+### master branch
+The [master](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master) branch hosts the continuous development of the AWS IoT Embedded C SDK libraries. Please be aware that the libraries in master branch may have bugs and performance issues. Consider using the released versions of the AWS IoT Embedded C SDK for production ready software.
+
+### v4_beta_deprecated branch
+The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v4_beta_deprecated) branch contains a beta version of the AWS IoT Embedded C SDK libraries, which is now deprecated. The libraries in this branch will not be released. However, critical bugs will be fixed and tested. No new features will be added to this branch.
+
+
+## Releases
+
+All the released versions of the AWS IoT Embedded C SDK libraries are available as git tags. For example the latest v3 SDK version is available at [tag 3.1.0](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v3.1.0).
```
Comment (muneebahmed10) [inline README.md:17]:
```suggestion
All of the released versions of the AWS IoT Embedded C SDK libraries are available as git tags. For example, the latest v3 SDK version is available at [tag 3.1.0](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v3.1.0).
```

File: README.md [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
 
 ## Building and Running Demos
 
   
 
-This SDK builds with [CMake](https://cmake.org/), a cross-platform build tool.
+The libraries in this SDK are not dependent on any Operating Systems. However, the demos for the libraries in this SDK are built and tested on Linux platform. This SDK builds with [CMake](https://cmake.org/), a cross-platform build tool.
```
Comment (muneebahmed10) [inline README.md:23]: Does Operating Systems need to be capitalized? Also,
```suggestion
The libraries in this SDK are not dependent on any Operating Systems. However, the demos for the libraries in this SDK are built and tested on a Linux platform. This SDK builds with [CMake](https://cmake.org/), a cross-platform build tool.
```

Comment (muneebahmed10) [inline README.md:23]:
```suggestion
The libraries in this SDK are not dependent on any operating systems. However, the demos for the libraries in this SDK are built and tested on a Linux platform. This SDK builds with [CMake](https://cmake.org/), a cross-platform build tool.
```

**Review by aggarg — DISMISSED** [review]

**Review by muneebahmed10 — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**